proposals/003-installer/installer-guidelines.md

```diff
@@ -0,0 +1,79 @@
+# Installer Guidelines
+
+## 0. Objectives
+
+We want people to use Haskell, which means the path to installing Haskell for first-time users
+should be short and easy, and it should yield an environment that is easy to use with the supporting
+literature. In addition, building simple programmes with a fresh installation should be as quick
+and painless as possible.
+
+These guidelines will set out the criteria that the Haskell Foundation will use in evaluating
+installers. We do not expect any individual to meet all these criteria initially but we would like
+them to be met by a collection of installers as quickly as possible.
+
+## 1. Cross Platform Coverage
+
+As many platforms should be covered by the installers as possible. For now, the minimum platforms
+that should be covered should include 64-bit Windows, macOS and Linux running on
+Intel-x86-compatible processors. This set of minimal platforms will soon have to be extended to
+include ARM-architecture processors.
+
+## 2. Uniformity of Delivered Environment
+
+Each installer should deliver a Haskell development environment that includes the following.
+
+  * The current recommended GHC toolchain, as determined by the installer, should be installed out
+    of the box, ready to use by all of the installed build systems.
+
+  * Both of the principal build systems (`cabal-install` and `stack`) should be installed,
+    available and ready for use in the way their associated development communities would expect.
+    (If an installer interrogates the user on installation about which components to install it is
```

Comment:
This is overly specific and aggressive UX marketing. Let the tool maintainers decide what reasonable defaults are.

Reply:
I disagree here.  It's sufficiently important to make both tools available as part of the standard Haskell environment that any tool linked to by the community as a general-purpose installer for Haskell should be installing both by default.  I'd add, though, that they should also be putting `ghc` and `ghci` in the path to be run directly, since that is *also* an important part of a standard Haskell environment.

This language still allows for tools to provide detailed options or customization.  It just says that if you install "Haskell", that includes working cabal and stack executables.

Reply:
Since stack today doesn't integrate well with globally installed GHC versions and has a self-update mechanism that can break your installer, I consider this feature experimental in ghcup and default installation of stack doesn't make sense yet.